rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -136,15 +139,20 @@ If the resulting list of transactions is empty and the `LC` value in the message
 
 #### 5.2.3 Transaction List
 
-When a node receives a `TransactionListQuery` message, it MUST respond with a `TransactionList` message.
+When a node receives a `TransactionListQuery` message, it SHOULD respond with a `TransactionList` message.
 This is a response type message so it MUST include the sent `conversationID`.
 Transactions that resulted from a `TransactionListQuery` MUST have been present in that message.
 Unknown transactions references SHOULD be ignored.
 Transactions that resulted from a `TransactionRangeQuery` MUST have an LC value that is within the requested range.
 If any of these requirements are not met, the entire message MUST be ignored.
+If the resulting list of transactions is empty, a node MAY ignore the message and not send a `TransactionList` message.
+If the list is not empty, it MUST send a response.
+All transactions in the `TransactionList` message MUST be sorted by LC value (lowest first).
+All transactions without a `pal` header MUST be added with their payload. Transactions with a `pal` header are discussed in [§7](rfc017-distributed-network-grpc-v2.md#7-private-transactions).
```

Comment:
should an error be returned?

Reply:
That would be ok I guess.

Reply:
`missing payload`

Reply:
Actually, if most of the messages are to be handled asynchronously, it'll not be possible to return an error!

Reply:
Since we use gRPC streaming, it is actually _never_ request-reply. It's many-requests, many replies. But you can still have an application-level request-reply protocol (like the conversations we now have), async or not?